### DISCUSSION TITLE
WEMOS LOLIN D32 pro, SD_Karte, TFT-Display

### DISCUSSION THREAD

**Discussion by kalus-pankow** [micropython/micropython#15412] [ESP32]
Hallo,
Vorweg, bin Anfänger!
 
Aus einer **Schleife** wird eine SD Karte aufgerufen um Daten zu lesen und zu speichern.
Anschließend wird ein TFT-Display aufgerufen.
das ganze ist in Micropython geschrieben auf einen WEMOS LOLIN D32 pro.
Vor der Schleife initialisiere ich:
**sd = SDCard(slot=3, sck=Pin(18), mosi=Pin(23), miso=Pin(19), cs=Pin(4))**
os.mount(sd, "/sd")

Ohne Aufruf des Displays läuft es problemlos
Wenn ich das Display aufrufe:
tft_CS = 19    
tft_RESET= 33    
tft_A0= 5
    
**spi = SPI(2, baudrate=20000000,mosi=Pin(13) , sck=Pin(14))  #miso=Pin(15)
tft=TFT(spi, tft_A0, tft_RESET, ft_CS)**

Werden die Daten richtig auf der SD-Karte verarbeitet und  auch angezeigt aber bei einem erneuten Aufruf der SD-Karte erfolgt die Fehlermeldung: 
**OSError: [Errno 2] ENOENT**

Wennich ich die SD-Karte am Anfang der Schleife initialisiere, kommt beim nächsten Schleifendurchlauf:
**OSError: (-259, 'ESP_ERR_INVALID_STATE')**

Wo liegt der Fehler?

**Comment by robert-hh** [comment]
A collision in the pin 19 use for both sd miso and tft cs.

**Comment by kalus-pankow** [comment]
Hallo,
danke für die schnelle Antwort!
Ja, das hätte ich auch als Anfänger sehen müssen!
Manchmal sieht man aber den Wald vor lauter Bäumen nicht.
Habe den Pin 15 genommen und siehe da, wenn man es richtig macht läuft es
auch.
Danke nochmals
Klaus

Am Fr., 5. Juli 2024 um 16:15 Uhr schrieb Robert Hammelrath <
***@***.***>:

> A collision in the pin 19 use for both sd miso and tft cs.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/micropython/discussions/15412#discussioncomment-9969051>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUFTY2SQ4LM4QWU6AQF5QODZK2S7NAVCNFSM6AAAAABKNCHKCCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TSNRZGA2TC>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>


-- 
Mit freundlichen Grüßen

Klaus-Dieter Schmidt

Telefon: 0179 4274007
E-Mail: ***@***.***
Gartenstr. 32
13127 Berlin
